doc/devices/dts/device_tree.rst

```diff
@@ -80,6 +80,54 @@ This .fixup file by default resides in the board directory and is named
 dts.fixup.  This fixup file maps the generated include information to the
 current driver/source usage.
 
+.. _dt_vs_kconfig:
+
+Device Tree vs Kconfig
+**********************
+
+As mentioned above there are several tools used to configuring the build with
+Zephyr.
+The two main ones, Device Tree and Kconfig might seem to overlap in purpose,
+but in fact they do not. This section is intended to serve as a reference when
+deciding whether to place a certain configuration item in Device Tree or Kconfig.
+
+The scope of each configuration tool can be summarized as follows:
+
+* Device Tree is used to describe the **hardware** and its **boot-time
```

Comment:
or build-time

Reply:
actually, this might be a bit confusing, since Kconfig is also determining the build-time configuration.
So maybe 
Device Tree is used to describe the hardware capabilities and configuration at build time.

Reply:
Well actually I disagree, I think my original wording (taken form Erwan) is clearer. Boot-time here really means that, when the hardware boots. Compilation and build time are really irrelevant IMHO

Reply:
OK, I see the "its" refers to "hardware, This clarifies it. 
In principle, "software" also has "boot-time" configuration.